### PR TITLE
Add `undef()` keyword to core 2.12 porting guide

### LINKED ISSUE #76382: Add undef() to the 2.12 porting guide
### Summary

https://github.com/ansible/ansible/pull/75435 added a new global Jinja function, which people may need to account for in their templating if they were using `undef` for other purposes.

For example, this template (simplified from an example that came up on IRC):
```
{% macro funroll_loops(parent=undef) %}
{% if parent is defined %}
{{ parent }}
{% endif %}
{% endmacro %}
{{ funroll_loops() }}
```

In old versions of Ansible the output file would be empty. In 2.12, it is:
```
<bound method Templar._make_undefined of <ansible.template.Templar object at 0x7f668e0ead00>>
```

### Issue Type

Documentation Report

### Component Name

porting_guide_core_2.12.rst

### Ansible Version

```console
$ ansible --version
ansible [core 2.12.0]
  config file = /home/ec2-user/ansible-aws/ansible/ansible.cfg
  configured module search path = ['/home/ec2-user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ec2-user/ansible-aws/.venv/lib/python3.8/site-packages/ansible
  ansible collection location = /home/ec2-user/ansible-aws/ansible/collections
  executable location = /home/ec2-user/ansible-aws/.venv/bin/ansible
  python version = 3.8.5 (default, Feb 18 2021, 01:24:20) [GCC 7.3.1 20180712 (Red Hat 7.3.1-12)]
  jinja version = 3.0.3
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed
```

### OS / Environment

n/a

### Additional Information

2.12 also hangs indefinitely with the following task, which I don't have time to dig into at the moment but is very odd behaviour:

```
    - debug:
        msg: "{{ undef | default('') }}"
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by lilatomic**
##### SUMMARY

fixes #76382

In Ansible 2.12, we added the `undef` function to the Jinja environment. This breaks other uses of `undef`, especially those expecting it to be undefined. This adds a mention in the porting guide.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/porting_guides/porting_guide_core_2.12.rst

##### ADDITIONAL INFORMATION

I built the docs, nothing seems to have caught fire.

### CODE REVIEW

**Review by mkrizek — APPROVED** [review]

**Comment by mkrizek** [comment]
Waiting for review from docs team.

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @lilatomic  for the Ansible docs fix! We talk about Ansible documentation on maxtrix at [#docs:ansible.im](https://matrix.to/#/#docs:ansible.im) and on libera IRC at #ansible-docs if you ever want to join us and chat about the docs!

**Pull request merged**
